pilot/pkg/networking/deprecated/ingress.go

```diff
@@ -41,22 +49,60 @@ func buildIngressListeners(mesh *meshconfig.MeshConfig, proxyInstances []*model.
 		store:            config,
 	}
 
-	listeners := []*xdsapi.Listener{buildHTTPListener(opts)}
+	manager := buildHTTPConnectionManager(opts)
+	l := newHTTPListener(opts.ip, opts.port, filterHTTPConnectionManager, messageToStruct(manager))
+	lc.HTTPListeners[":80"] = l
+
+	listeners := []*xdsapi.Listener{l}
 
 	// lack of SNI in Envoy implies that TLS secrets are attached to listeners
 	// therefore, we should first check that TLS endpoint is needed before shipping TLS listener
 	_, secret := v1.BuildIngressRoutes(mesh, ingress, proxyInstances, discovery, config)
 	if secret != "" {
 		opts.port = 443
 		opts.rds = "443"
-		listener := buildHTTPListener(opts)
-		// TODO(mostrowski)
-		/*listener.SSLContext = &SSLContext{
-			CertChainFile:  path.Join(model.IngressCertsPath, model.IngressCertFilename),
-			PrivateKeyFile: path.Join(model.IngressCertsPath, model.IngressKeyFilename),
-			ALPNProtocols:  strings.Join(ListenersALPNProtocols, ","),
-		} */
-		listeners = append(listeners, listener)
+		manager := buildHTTPConnectionManager(opts)
+		l := newHTTPListener(opts.ip, opts.port, filterHTTPConnectionManager, messageToStruct(manager))
+		lc.HTTPListeners[":443"] = l
+
+		l.FilterChains = []listener.FilterChain{
+			{
+				TlsContext: &envoy_api_v2_auth.DownstreamTlsContext{
+					CommonTlsContext: &envoy_api_v2_auth.CommonTlsContext{
+						ValidationContext: &envoy_api_v2_auth.CertificateValidationContext{
+							TrustedCa: &core.DataSource{
+								Specifier: &core.DataSource_InlineString{
+									InlineString: "./security/samples/plugin_ca_certs/cert-chain.pem",
+								},
+							},
+						},
+						AlpnProtocols: v1.ListenersALPNProtocols,
```

Comment:
copy this constant into networking/v1alpha3/.. please and use that

Reply:
The copying is because we eventually want to do a full check of the code to see what else is being borrowed from v1. I don't mind copying to some shared place, but I dont think costin agrees with that.

Reply:
We can just as easily do that check by grepping for "v1." or removing the import and seeing what breaks. This just seems like a straightforward case of "Don't Repeat Yourself".